patches/jsc.patch

```diff
@@ -291,3 +291,58 @@ diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/S
  )
 +add_definitions(-DJSC_VERSION="${JSC_VERSION}")
 
+diff -aur target-org/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp target/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp
+--- target-org/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp	2018-08-24 16:08:22.000000000 +0800
++++ target/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp	2018-11-07 22:59:05.000000000 +0800
+@@ -38,6 +38,30 @@
+ #if OS(LINUX)
+ static const Seconds s_memoryFootprintUpdateInterval = 1_s;
+
++#if defined(__ANDROID__) && __ANDROID_API__ < 18
++//
++// The POSIX.1 C getline isn't exported until API 18 in NDK.
++// As the getline() used to read content from /proc/self/smaps,
++// the content fortunately are plain text and lines are not too long.
++// Here to simply use fgets() as fallback.
++// Some differences further:
++//  1. Line buffer is fixed 512 bytes.
++//  2. Return 1 if succeed.
++//     This is only for forEachLine() use.
++//     Just don't want to do strlen() calculating length again.
++//
++static ssize_t getline(char** linep, size_t* linecapp, FILE* stream) {
++  int fixedBufferSize = 512;
++  if (*linep == nullptr) {
++    *linep = reinterpret_cast<char*>(malloc(fixedBufferSize));
++  }
++  if (fgets(*linep, fixedBufferSize, stream) == nullptr)
++    return -1;
++
++  return 1;
++}
++#endif
++
+ template<typename Functor>
+ static void forEachLine(FILE* file, Functor functor)
+ {
+diff -aur target-org/webkit/Source/JavaScriptCore/Sources.txt target/webkit/Source/JavaScriptCore/Sources.txt
+--- target-org/webkit/Source/JavaScriptCore/Sources.txt 2018-11-11 23:05:40.000000000 +0800
++++ target/webkit/Source/JavaScriptCore/Sources.txt     2018-11-12 00:03:26.000000000 +0800
+@@ -1046,3 +1046,6 @@
+
+ // Derived Sources
+ yarr/YarrCanonicalizeUnicode.cpp
++
++// Polyfills
++polyfills/log2.cpp
+diff -aur /dev/null target/webkit/Source/JavaScriptCore/polyfills/log2.cpp
+--- /dev/null   2018-11-12 01:21:57.000000000 +0800
++++ target/webkit/Source/JavaScriptCore/polyfills/log2.cpp      2018-11-12 01:19:49.000000000 +0800
+@@ -0,0 +1,7 @@
++#include <math.h>
++
++#if defined(__ANDROID__) && __ANDROID_API__ < 19
```

Comment:
```suggestion
+#if defined(__ANDROID__) && __ANDROID_API__ < 18
```

This does not have any implications for now but `log2` is available since API 18

Reply:
Oops, my fault of . Thank you for the finding.